### PR TITLE
Added greater and less operators

### PR DESCRIPTION
**Pull request by turneram**

### CODE REVIEW

File: src/onnx/onnx.cpp [hunk 1]
```diff
@@ -2486,6 +2488,28 @@ struct onnx_parser
...
+    parse_less(const std::string&, const node_info&, std::vector<instruction_ref> args)
+    {
+        auto l = add_broadcastable_binary_op(args[0], args[1], "less");
+        if(l->get_shape().type() != shape::bool_type)
+        {
+            l = prog.add_instruction(make_op("convert", {{"target_type", shape::bool_type}}), l);
+        }
+        return l;
+    }
+
```
Comment (scxiao) [inline src/onnx/onnx.cpp:2512]: the combined function would be like:

instruction_ref parse_compare_op(const std::string&,
                             const std::string& op_name,
                             const node_info&,
                             std::vector<instruction_ref> args)
{
    auto l = add_broadcastable_binary_op(args[0], args[1], op_name);
    if(l->get_shape().type() != shape::bool_type)
    {
        l = prog.add_instruction(make_op("convert", {{"target_type", shape::bool_type}}), l);
    }
    return l;
}

Then the add_mem_op is called like:

add_mem_op("Equal", "equal", &onnx_parser::parse_compare_op);
add_mem_op("Greater", "greater", &onnx_parser::parse_compare_op);
add_mem_op("Less", "less", &onnx_parser::parse_compare_op);

Comment (kahmed10) [reply src/onnx/onnx.cpp]: this is also the main suggested code change I had.

File: src/include/migraphx/op/greater.hpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct greater : binary<greater>
+{
+    value attributes() const
+    {
+        auto a           = base_attributes();
+        a["commutative"] = false;
```
Comment (pfultz2) [inline src/include/migraphx/op/greater.hpp:18]: Greater operator is not commutative.

File: src/include/migraphx/op/less.hpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct less : binary<less>
+{
+    value attributes() const
+    {
+        auto a           = base_attributes();
+        a["commutative"] = false;
```
Comment (pfultz2) [inline src/include/migraphx/op/less.hpp:18]: Less operator is not commutative.

File: src/include/migraphx/op/greater.hpp (hunk 2 above)
Comment (pfultz2) [reply src/include/migraphx/op/greater.hpp:18]: This function should just be removed.

File: src/include/migraphx/op/less.hpp (hunk 3 above)
Comment (pfultz2) [reply src/include/migraphx/op/less.hpp:18]: This function should just be removed.

Comment (turneram) [reply src/include/migraphx/op/less.hpp]: Removed

File: src/include/migraphx/op/greater.hpp (hunk 2 above)
Comment (turneram) [reply src/include/migraphx/op/greater.hpp]: Removed

File: src/onnx/onnx.cpp (hunk 1 above)
Comment (turneram) [reply src/onnx/onnx.cpp]: Added

File: src/onnx/onnx.cpp [hunk 4]
```diff
@@ -135,16 +135,18 @@ struct onnx_parser
         add_mem_op("ConvTranspose", &onnx_parser::parse_conv_transpose);
         add_mem_op("Dropout", &onnx_parser::parse_dropout);
         add_mem_op("Elu", &onnx_parser::parse_elu);
-        add_mem_op("Equal", &onnx_parser::parse_equal);
+        add_mem_op("Equal", &onnx_parser::parse_compare_op);
```
Comment (scxiao) [inline src/onnx/onnx.cpp:138]: I think there should be another input for this function call, which is:
add_mem_op("Equal", "equal", &onnx_parser::parse_compare_op);

Comment (turneram) [reply src/onnx/onnx.cpp]: Added

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (kahmed10) [inline .vscode/settings.json:1]: this file should be removed from the commit

File: .gitignore [hunk 6]
```diff
@@ -1 +1,3 @@
 *.swp
+**/build
```
Comment (kahmed10) [inline .gitignore:2]: this file should be removed from the commit

File: .vscode/settings.json (hunk 5 above)
Comment (turneram) [reply .vscode/settings.json]: Removed

File: .gitignore (hunk 6 above)
Comment (turneram) [reply .gitignore]: Fixed

File: test/cpu_ops_test.cpp [hunk 7]
```diff
@@ -3006,4 +3006,86 @@ TEST_CASE(equal_brcst_test)
...
+        p.add_literal(migraphx::literal{s, {1.1, 1.6, -0.1, -1.2, -1.5, -0.7, 0.0, 2.3, -2.1}});
+    auto le = p.add_instruction(migraphx::op::less{}, l0, l1);
+    auto r  = p.add_instruction(migraphx::op::convert{migraphx::shape::bool_type}, le);
+    p.add_return({r});
+
+    p.compile(migraphx::cpu::target{});
+    auto result = p.eval({}).back();
+    std::vector<bool> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+    std::vector<bool> gold = {0, 1, 0, 0, 0, 0, 0, 1, 0};
```
Comment (kahmed10) [inline test/cpu_ops_test.cpp:3066]: this needs to be changed from 0s and 1s to true and false literals.

File: test/cpu_ops_test.cpp [hunk 8]
```diff
@@ -3006,4 +3006,86 @@ TEST_CASE(equal_brcst_test)
...
+    auto bl1 = p.add_instruction(migraphx::op::multibroadcast{{3, 3}}, l1);
+    auto le  = p.add_instruction(migraphx::op::less{}, l0, bl1);
+    auto r   = p.add_instruction(migraphx::op::convert{migraphx::shape::bool_type}, le);
+    p.add_return({r});
+
+    p.compile(migraphx::cpu::target{});
+    auto result = p.eval({}).back();
+    std::vector<bool> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+    std::vector<bool> gold = {0, 0, 1, 0, 0, 0, 0, 0, 1};
```
Comment (kahmed10) [inline test/cpu_ops_test.cpp:3087]: same as above

File: test/cpu_ops_test.cpp (hunk 7 above)
Comment (turneram) [reply test/cpu_ops_test.cpp]: Fixed

File: test/cpu_ops_test.cpp (hunk 8 above)
Comment (turneram) [reply test/cpu_ops_test.cpp]: Fixed

File: src/onnx/onnx.cpp [hunk 9]
```diff
@@ -2906,4 +2910,4 @@ program parse_onnx_buffer(const void* data, std::size_t size, const onnx_options
 }
 
 } // namespace MIGRAPHX_INLINE_NS
-} // namespace migraphx
+} // namespace migraphx
```
Comment (kahmed10) [inline src/onnx/onnx.cpp:2913]: There should be a newline at the end of the file.

**Review by kahmed10 — APPROVED** [review]

**Review by scxiao — APPROVED** [review]

**Pull request merged**